backend/JavaWebService/src/main/java/com/kainos/ea/resources/JobSpecificationController.java

```diff
@@ -0,0 +1,30 @@
+package com.kainos.ea.resources;
+
+import com.kainos.ea.db.RoleMapper;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+@Path("/api")
+public class JobSpecificationController {
+    private JobSpecificationService service;
+
+    public JobSpecificationController(JobSpecificationService service){
+        this.service = service;
+    }
+
+    @GET
+    @Path("/job-roles/job-spec/{id}")
```

Comment:
Job spec shouldn't need it's own url - it should use the job-roles/{id} endpoint

Reply:
RESTful APIs follow a consistent pattern
`GET /resources` --> retrieve all resources
`GET /resource/id` --> pick the resource with this id
`GET /resource/id/sub-resource` --> retrieve all subresources from the resource with this id
`GET /resource/id/sub-resource/sub-id` --> pick the subresource with the sub-id from the resource with this id
... and so on

Filters and sorting etc would normally be defined by parameters. So
`GET /courses/3/students?sort=surname` would fetch all of the students for the course with ID 3, and sort them by their last names.